### PR TITLE
feat: introducing react query

### LINKED ISSUE #62: Refactor frontend to use react query

### PR DESCRIPTION
**Pull request by Mautjee**
Introduction of react query to the frontend 

closes #62

### CODE REVIEW

File: k8s/local/skaffold.local.full.yaml [hunk 1]
```diff
@@ -0,0 +1,172 @@
+apiVersion: skaffold/v4beta6
```
Comment (FilipHarald) [inline k8s/local/skaffold.local.full.yaml:1]: Great idea! I think this should be the default one to help potential new devs to get started.

The other one you can name `k8s/local/skaffold.local.light.yaml`.

Also: make sure to update the README!

File: services/explorer-ui/package.json [hunk 2]
```diff
@@ -26,9 +26,10 @@
     "@radix-ui/react-tabs": "1.0.4",
     "@radix-ui/react-toast": "1.1.5",
     "@radix-ui/react-tooltip": "1.0.7",
-    "@tanstack/react-query": "5.32.0",
+    "@tanstack/react-query": "^5.56.2",
```
Comment (FilipHarald) [inline services/explorer-ui/package.json:29]:
```suggestion
    "@tanstack/react-query": "5.56.2",
```

File: services/explorer-ui/package.json [hunk 3]
```diff
@@ -26,9 +26,10 @@
...
     "@tanstack/react-router": "1.58.3",
     "@tanstack/react-table": "8.16.0",
+    "axios": "^1.7.7",
```
Comment (FilipHarald) [inline services/explorer-ui/package.json:32]:
```suggestion
    "axios": "1.7.7",
```

File: services/explorer-ui/src/api/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export * from "./contract";
+export * from "./block";
+export * from "./tx-effects";
```
Comment (FilipHarald) [inline services/explorer-ui/src/api/index.ts:3]: This should be singular like the other files. (Or all of them should be plural)

`tx-effect`

File: services/explorer-ui/src/hooks/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export * from "./useBlock";
```
Comment (FilipHarald) [inline services/explorer-ui/src/hooks/index.ts:1]: If you are going to have multiple hooks in this file I think it's better to rename the file to `services/explorer-ui/src/hooks/block.ts`

File: yarn.lock [hunk 6]
```diff
@@ -972,7 +972,7 @@ __metadata:
     "@radix-ui/react-toast": "npm:1.1.5"
     "@radix-ui/react-tooltip": "npm:1.0.7"
     "@tailwindcss/typography": "npm:0.5.10"
-    "@tanstack/react-query": "npm:5.32.0"
+    "@tanstack/react-query": "npm:^5.56.2"
```
Comment (FilipHarald) [inline yarn.lock:975]:
```suggestion
    "@tanstack/react-query": "npm:5.56.2"
```

File: yarn.lock [hunk 7]
```diff
@@ -986,6 +986,7 @@ __metadata:
     "@typescript-eslint/parser": "npm:7.15.0"
     "@vitejs/plugin-react": "npm:4.3.1"
     autoprefixer: "npm:10.4.19"
+    axios: "npm:^1.7.7"
```
Comment (FilipHarald) [inline yarn.lock:989]:
```suggestion
    axios: "npm:1.7.7"
```

File: k8s/local/skaffold.local.full.yaml (hunk 1 above)
Comment (Mautjee) [reply k8s/local/skaffold.local.full.yaml:1]: Why change the name to light?

Comment (FilipHarald) [reply k8s/local/skaffold.local.full.yaml:1]: I think we should have these two:

`skaffold.local.yaml` <- default (full)
`skaffold.local.light.yaml` <- for a lighter restart. You can take another name, like "small", if you like that better.

File: README.md [hunk 8]
```diff
@@ -23,12 +29,11 @@ _Make sure you have .chicmoz-local.env file in the root_
...
-terminal 3:
+## Pro tip
 
-```sh
-cd services/explorer-ui && yarn dev
-```
+when developing locally you can command out the creation of the `explorer-ui` and run it locally
 
-- UI: http://localhost:5173
-- API: https://explorer-api.localhost:443
```
Comment (FilipHarald) [inline README.md:33]: I think we should still have this

File: README.md [hunk 9]
```diff
@@ -14,6 +14,12 @@ terminal 1:
 minikube start --kubernetes-version=v1.25.3 --cpus max --memory max && skaffold run
 ```
 
+after the first time you have deployed the app you can run the following command to have faster deployments:
```
Comment (FilipHarald) [inline README.md:17]: Let's put this under Pro tip also. So our main is only "terminal 1" and "terminal 2"

File: k8s/local/skaffold.local.ligth.yaml [hunk 10]
```diff
@@ -0,0 +1,94 @@
+apiVersion: skaffold/v4beta6
```
Comment (FilipHarald) [inline k8s/local/skaffold.local.ligth.yaml:1]: misspelled, change ligth => light

File: README.md [hunk 11]
```diff
@@ -23,12 +23,21 @@ _Make sure you have .chicmoz-local.env file in the root_
...
+```
+
+when developing locally you can command out the creation of the `explorer-ui` and run it locally
+
+```
+cd services/exporer-ui/
+yarn dev
 ```
 
 - UI: http://localhost:5173
```
Comment (FilipHarald) [inline README.md:41]: You can remove this

**Review by FilipHarald — APPROVED** [review]
with the last two comments it's good!

File: README.md [hunk 12]
```diff
@@ -23,12 +23,20 @@ _Make sure you have .chicmoz-local.env file in the root_
...
+```
+cd services/exporer-ui/
+yarn dev
 ```
 
-- UI: http://localhost:5173
-- API: https://explorer-api.localhost:443
-- (index of API: https://explorer-api.localhost:443/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
+- API: http://explorer-api.localhost
+  (index of API: http://explorer-api.localhost/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
```
Comment (FilipHarald) [inline README.md:42]:
```suggestion
  - (index of API: http://explorer-api.localhost/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
```

File: README.md [hunk 13]
```diff
@@ -23,12 +29,11 @@ _Make sure you have .chicmoz-local.env file in the root_
...
+## Pro tip
 
-```sh
-cd services/explorer-ui && yarn dev
-```
+when developing locally you can command out the creation of the `explorer-ui` and run it locally
 
-- UI: http://localhost:5173
-- API: https://explorer-api.localhost:443
-- (index of API: https://explorer-api.localhost:443/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
```
[the remark this answers is not in the document]
Comment (FilipHarald) [reply README.md:34]: Fixed here: https://github.com/aztlan-labs/chicmoz/pull/63/files#r1781209463

File: README.md [hunk 14]
```diff
@@ -23,12 +23,20 @@ _Make sure you have .chicmoz-local.env file in the root_
...
+```
+cd services/exporer-ui/
+yarn dev
 ```
 
-- UI: http://localhost:5173
-- API: https://explorer-api.localhost:443
-- (index of API: https://explorer-api.localhost:443/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
+- API: http://explorer-api.localhost
+  - (index of API: http://explorer-api.localhost/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
```
Comment (FilipHarald) [inline README.md:42]:
```suggestion
- (index of API: http://explorer-api.localhost/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
```

File: README.md [hunk 15]
```diff
@@ -23,12 +23,20 @@ _Make sure you have .chicmoz-local.env file in the root_
...
+```
+cd services/exporer-ui/
+yarn dev
 ```
 
-- UI: http://localhost:5173
-- API: https://explorer-api.localhost:443
-- (index of API: https://explorer-api.localhost:443/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
+- API: http://explorer-api.localhost
+- (index of API: http://explorer-api.localhost/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index)
```
Comment (FilipHarald) [inline README.md:42]:
```suggestion
- index of API: http://explorer-api.localhost/v1/d1e2083a-660c-4314-a6f2-1d42f4b944f4/l2/index
```

**Pull request merged**